### ISSUE TITLE
im getting this error in jupyter-  Exception encountered when calling Lambda.call() in this line- attention_context = Lambda(lambda x: tf.einsum('ijk,ijl->ikl', x[0], x[1]))([attention_scores, se3])

### ISSUE THREAD

**Issue by ShahidRRRRRR** [tensorflow/tensorflow#63795]
# Encoder model
inputs1 = Input(shape=(4096,))
fe1 = Dropout(0.5)(inputs1)
fe2 = Dense(256, activation='relu')(fe1)
fe2_projected = RepeatVector(max_caption_length)(fe2)
fe2_projected = Bidirectional(LSTM(256, return_sequences=True))(fe2_projected)

# Sequence feature layers
inputs2 = Input(shape=(max_caption_length,))
se1 = Embedding(vocab_size, 256, mask_zero=True)(inputs2)
se2 = Dropout(0.5)(se1)
se3 = Bidirectional(LSTM(256, return_sequences=True))(se2)

# Apply attention mechanism using Dot product
attention = Dot(axes=[2, 2])([fe2_projected, se3])  # Calculate attention scores

# Softmax attention scores
attention_scores = Activation('softmax')(attention)

# Apply attention scores to sequence embeddings
attention_context = Lambda(lambda x: tf.einsum('ijk,ijl->ikl', x[0], x[1]))([attention_scores, se3])

# Sum the attended sequence embeddings along the time axis
context_vector = tf.reduce_sum(attention_context, axis=1)

# Decoder model
decoder_input = concatenate([context_vector, fe2], axis=-1)
decoder1 = Dense(256, activation='relu')(decoder_input)
outputs = Dense(vocab_size, activation='softmax')(decoder1)

# Create the model
model = Model(inputs=[inputs1, inputs2], outputs=outputs)
model.compile(loss='categorical_crossentropy', optimizer='adam')

# Visualize the model
plot_model(model, show_shapes=True)

im getting this error someone help
Exception encountered when calling Lambda.call().

In a nested call() argument, you cannot mix tensors and non-tensors. Received invalid mixed argument: mask=[None, <KerasTensor shape=(None, 34), dtype=bool, sparse=False, name=keras_tensor_42>]

**Comment by Tanishq-JM** [comment]
It looks like the issue is related to the use of the `Lambda` layer in your code. The error message suggests that there's a mix of tensors and non-tensors in a nested `Lambda` call, specifically related to the `mask` argument.

The error might be occurring because the `Lambda` layer is expecting a tensor for its `mask` argument, but it's receiving a list with `None` and a tensor instead. To resolve this, you can try modifying the `Lambda` layer to explicitly handle the `mask` argument. Here's an updated version of the `Lambda` layer that should resolve the issue:

`attention_context = Lambda(lambda x: tf.einsum('ijk,ijl->ikl', x[0], x[1]), mask=lambda inputs, mask: mask[1])([attention_scores, se3])
`

In this modification, the `mask` argument is explicitly handled to pass the mask from the `se3` input to the `Lambda` layer.

Please try this modification and see if it resolves the issue. If the error persists, there may be other issues in your code that need to be addressed.

**Comment by sushreebarsa** [comment]
@ShahidRRRRRR Please print the shapes of attention_scores and se3 before passing them to the Lambda layer to verify compatibility. If you still face the issue then, could you please ensure to use the latest TF version and share the code in a notebook or colab gist? 

Thank you!
